### PR TITLE
Substitute author and email by git config

### PR DESCRIPTION
**Pull request by to-s**
# Fix/Feature/Other

## Fix

- [ ] Link to issue. If there is no issue please describe it using at least the issue template

https://github.com/cschlosser/doxdocgen/issues/182

- [ ] Description of the fix

- [ ] Added unit test

## Feature

- [ ] Description of what's added

Option to substitute {author} with git config --get user.name
Option to substitute {email} with git config --get user.email

- [ ] Gif of your feature if appropriate

- [ ] Added gif to README

- [ ] Added unit test

## Other

- [ ] Description

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -253,6 +263,7 @@
     "test": "npm run compile && node ./node_modules/vscode/bin/test"
   },
   "dependencies": {
+    "simple-git": "^2.20.1",
```
Comment (cschlosser) [inline package.json:266]: The currently used typescript version is too old for the new package.
#187 should address this or you switch to typescript 3.8.3 with this PR

File: src/Config.ts [hunk 2]
```diff
@@ -99,6 +102,18 @@ export class Config {
...
+
+        // Check if set to use the git username
+        if (values.Generic.useGitUserName === true) {
+            values.Generic.authorName = gitConfig.UserName.toString();
+        }
+
+        // Check if set to use the git email
+        if (values.Generic.useGitUserEmail === true) {
+            values.Generic.authorEmail = gitConfig.UserEmail.toString();
+        }
```
Comment (cschlosser) [inline src/Config.ts:116]: Please move this to `CppDocGen.generateAuthorTag`

File: package.json [hunk 3]
```diff
@@ -225,6 +225,16 @@
...
+        },
+        "doxdocgen.generic.useGitUserName": {
+          "description": "Substitute {author} with git config --get user.name.",
+          "type": "boolean",
+          "default": false
+        },
+        "doxdocgen.generic.useGitUserEmail": {
+          "description": "Substitute {email} with git config --get user.email.",
+          "type": "boolean",
+          "default": false
```
Comment (cschlosser) [inline package.json:237]: Please add these values to the README

**Review by cschlosser — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (to-s) [reply package.json]: Upgraded typescript in https://github.com/cschlosser/doxdocgen/pull/186/commits/f0a69da5fa72c06679b9feaf3a113842f695a503

File: src/Config.ts (hunk 2 above)
Comment (to-s) [reply src/Config.ts]: Moved with https://github.com/cschlosser/doxdocgen/pull/186/commits/0a5167e25756fa381dec2b8aa005077101217580

File: package.json (hunk 3 above)
Comment (to-s) [reply package.json:237]: Added by https://github.com/cschlosser/doxdocgen/pull/186/commits/94c96625b0a34db98def47957e33cb47ac93c3b7

**Review by cschlosser — APPROVED** [review]
Thank you.
That's looking good. Wait for CI to pass merge and then I will try a new release

**Pull request merged**